### PR TITLE
Add LiveZilla Tag

### PR DESCRIPTION
**Pull request by scysys**
Add LiveZilla Tag

Tested and working.
Using it a few days on own Websites.

I think it has all options that are needed.

### CODE REVIEW

File: Template/Tag/LivezillaDynamicTag.php [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        return array(
+            $this->makeSetting('LivezillaDynamicID', '', FieldConfig::TYPE_STRING, function (FieldConfig $field) {
+                $field->title = 'Livezilla ID';
+                $field->description = 'Insert the Livezilla ID from your Dynamic Code section.';
+                $field->validators[] = new NotEmpty();
+                $field->validators[] = new CharacterLength(32);
+            }),
+			$this->makeSetting('LivezillaDynamicDomain', '', FieldConfig::TYPE_STRING, function (FieldConfig $field) {
+                $field->title = 'Livezilla Domain';
+                $field->description = 'Enter your Domain without http:// or https://';
```
[the remark this answers is not in the document]
Comment (scysys) [reply Template/Tag/LivezillaDynamicTag.php]: Livezilla is an Chat and Monitoring solution. Thoose things violates privacy of an website visitor. I do not tolerate those implementations over an insecure http connection!

File: Template/Tag/LivezillaDynamicTag.web.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+(function () {
```
[the remark this answers is not in the document]
Comment (scysys) [reply Template/Tag/LivezillaDynamicTag.web.js]: Don´t find it on theire website. I took an Screenshot from the Client itself.

bildschirmfoto 2018-10-02 um 20 44 01

File: Template/Tag/LivezillaDynamicTag.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (Findus23) [reply Template/Tag/LivezillaDynamicTag.php]: I fully agree, then you should make it clear to the user (e.g. make them enter the URL and throw an error if it isn't HTTPS).

If you want to I can help you write this validate function.

File: Template/Tag/LivezillaDynamicTag.web.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Findus23) [reply Template/Tag/LivezillaDynamicTag.web.js]: Thanks this makes sense.

File: Template/Tag/LivezillaDynamicTag.php [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+                $field->validators[] = new CharacterLength(4, 60);
+            }),
+			$this->makeSetting('LivezillaDynamicDefer', 'true', FieldConfig::TYPE_STRING, function (FieldConfig $field) {
+                $field->title = 'Livezilla Script "defer"?';
+                $field->description = 'In most cases it should be the Standard. (true)
+				Available options are: true or false'; // Howto linebreak?
+				$field->validators[] = new NotEmpty();
+                $field->validators[] = new CharacterLength(4, 5);
+            }),
+			$this->makeSetting('LivezillaDynamicCookieLawName', '', FieldConfig::TYPE_STRING, function (FieldConfig $field) {
```
[the remark this answers is not in the document]
Comment (scysys) [reply Template/Tag/LivezillaDynamicTag.php]: I using it with an other implantation for the tag-manager. It respect the Privacy of European Users and load this Tag only when he really accept those parts on an website.

The tag-manager gives me no real way to document this for other users.

File: lang/de.json [hunk 4]
```diff
@@ -278,6 +278,9 @@
         "LastUpdated": "Zuletzt aktualisiert",
         "LastVersions": "Letzte Versionen",
         "LearnMore": "Mehr dazu",
+		"LivezillaDynamicTagName": "Livezilla Chat / Monitoring",
```
[the remark this answers is not in the document]
Comment (scysys) [reply lang/de.json]: I did not knew that.

File: Template/Tag/LivezillaDynamicTag.php [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+    public function getParameters() {
+        return array(
+            $this->makeSetting('LivezillaDynamicID', '', FieldConfig::TYPE_STRING, function (FieldConfig $field) {
+                $field->title = 'Livezilla ID';
+                $field->description = 'Insert the Livezilla ID from your Dynamic Code section.';
+                $field->validators[] = new NotEmpty();
+                $field->validators[] = new CharacterLength(32);
+            }),
+			$this->makeSetting('LivezillaDynamicDomain', '', FieldConfig::TYPE_STRING, function (FieldConfig $field) {
```
Comment (tsteur) [inline Template/Tag/LivezillaDynamicTag.php]: I'm not quite sure but could this be maybe automatically configured in the JavaScript depending on the current domain? eg `var LivezilleDomain = localhost.origin` or so?

Comment (Findus23) [reply Template/Tag/LivezillaDynamicTag.php]: @tsteur I am not sure, but as I understood it this is the URL where Livezilla is installed (so similar to the Matomo URL)

Comment (tsteur) [reply Template/Tag/LivezillaDynamicTag.php]: I see. Makes sense 👍

File: Template/Tag/LivezillaDynamicTag.php [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+                $field->title = 'Livezilla Domain';
+                $field->description = 'Enter the Domain where Livezilla is installed. Example: https://www.example.com';
+                $field->validators[] = new NotEmpty();
+				$field->validators[] = new UrlLike();
+                $field->validators[] = new CharacterLength(11, 60);
+            }),
+			$this->makeSetting('LivezillaDynamicDefer', true, FieldConfig::TYPE_BOOL, function (FieldConfig $field) {
+                $field->title = 'Livezilla Script "defer"?';
+				$field->uiControl = FieldConfig::UI_CONTROL_CHECKBOX;
+                $field->description = 'In most cases you should let it activated.';
```
Comment (tsteur) [inline Template/Tag/LivezillaDynamicTag.php]: If possible, be good to give an example when it makes sense to disable it. Not needed though.

**Comment by tsteur** [comment]
@scysys @Findus23 looks good to me. I would only replace all tabs with spaces and sometimes it looks like there are two many tabs/spaces.

**Comment by tsteur** [comment]
Tested it and worked fine for me.

File: Template/Tag/LivezillaDynamicTag.php (hunk 6 above)
Comment (scysys) [reply Template/Tag/LivezillaDynamicTag.php]: No its really not needed. But sometimes you have sites where you would remove this part. So there are 2 Options. Remove "defer" completely or leave it as default and give an user no option s to remove it.

File: Template/Tag/LivezillaDynamicTag.php [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+    {
+        // You may optionally specify a path to an image icon URL, for example:
+        //
+        // return 'plugins/TagManager/images/MyIcon.png';
+        //
+        // to not return default icon call:
+        // return parent::getIcon();
+        //
+        // The image should have ideally a resolution of about 64x64 pixels.
+        return 'plugins/TagManager/images/icons/livezilla_icon.png';
```
[the remark this answers is not in the document]
Comment (scysys) [reply Template/Tag/LivezillaDynamicTag.php]: I try to find another Image, but im failed :(

File: Template/Tag/LivezillaDynamicTag.php (hunk 6 above)
Comment (scysys) [reply Template/Tag/LivezillaDynamicTag.php]: An really better solution would be an general option in Tag Manager to async or defer scripts. I don't find those options at this moment.

Example from Qubit´s OpenTag Manager
bildschirmfoto 2018-10-05 um 13 51 28

**Pull request merged**